catalog/db_scanner.go

```diff
@@ -7,6 +7,7 @@ type DBScannerOptions struct {
 	After            string
 	AdditionalFields []string
 	AdditionalWhere  sq.Sqlizer
+	TrimmedLineage   []lineageCommit
```

Comment:
The DB scanner options are common options used by both Scanners - using lineage as part of options that we pass to branch scanner just break the 'common' idea.
Also the term TrimmedLineage - is the implementation detail of a specific use-case, we should keep the scanner option general as much as possible.

In this case, the DBLineageScanner need to accept LineageScannerOptions that has Lineage as an optional field - this will enable us to create scanner and override the default lineage.

Reply:
Done